sdk/core/Azure.Core/src/Pipeline/HttpPipelinePolicy.cs

```diff
@@ -2,25 +2,28 @@
 // Licensed under the MIT License.
 
 using System;
+using System.ClientModel.Primitives;
+using System.Collections;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace Azure.Core.Pipeline
 {
     /// <summary>
     /// Represent an extension point for the <see cref="HttpPipeline"/> that can mutate the <see cref="Request"/> and react to received <see cref="Response"/>.
     /// </summary>
-    public abstract class HttpPipelinePolicy
+    public abstract class HttpPipelinePolicy : PipelinePolicy
```

Comment:
Since we are doing adapters, is the inheritance needed?

Reply:
The ClientModel pipeline needs to be able to call ProcessNext on an Azure.Core policy from a System.ClientModel policy.  We can look at another way of doing this, but inheritance allows us to do this rather simply.